### PR TITLE
Refactor diag_manager-controlled diagnostics for backwards compatibility

### PR DESCRIPTION
**Pull request by spencerkclark**
This PR refactors the diagnostics-manager-controlled diagnostics introduced in #112 such that they are defined within the same data structure as the other physics diagnostics.  It does so by simply adding a flag to the derived data type (default `.false.`) denoting whether a particular diagnostic is `diag_manager_controlled` or not.  The code path taken for sending data to the diagnostics manager then depends on the value of this flag.  It should not change any answers or functionality (regression tests confirm this).

xref: https://github.com/VulcanClimateModeling/fv3gfs-wrapper/issues/230

### CODE REVIEW

File: FV3/gfsphysics/GFS_layer/GFS_diagnostics.F90 [hunk 1]
```diff
@@ -3614,7 +3268,7 @@ subroutine GFS_externaldiag_populate (ExtDiag, Model, Statein, Stateout, Sfcprop
     idx = idx + 1
     ExtDiag(idx)%axes = 2
     ExtDiag(idx)%name = 'ifd'
-    ExtDiag(idx)%desc = 'nsst index to start dtlm run or not'
+    ExtDiag(idx)%desc = 'nsst idx to start dtlm run or not'
```
Comment (oliverwm1) [inline FV3/gfsphysics/GFS_layer/GFS_diagnostics.F90:3271]: is this change intentional?

File: FV3/io/FV3GFS_io.F90 [hunk 2]
```diff
@@ -3008,7 +2993,8 @@ subroutine send_diag_manager_controlled_diagnostic_data(Time, Diag, Atm_block, I
 
     do index = 1, DIAG_SIZE
       if (trim(Diag(index)%name) .eq. '') exit
-      if (Diag(index)%id .gt. 0 .or. Diag_coarse(index)%id .gt. 0) then
+      requested = Diag(index)%id .gt. 0 .or. Diag_coarse(index)%id .gt. 0
+      if (requested .and. Diag(index)%diag_manager_controlled) then
```
Comment (oliverwm1) [inline FV3/io/FV3GFS_io.F90:2997]: I'm probably not following correctly... but what if `diag_manager_controlled` is not defined? It looked like some of the diagnostics in `GFS_diagnostics` have it set to true, but for others it's not defined. Put another way, where is the default set?

File: FV3/io/FV3GFS_io.F90 [hunk 3]
```diff
@@ -3139,7 +3126,8 @@ subroutine fv3gfs_diag_output(time, diag, atm_block, IPD_Data, nx, ny, levs, ntc
      
 !     if(mpp_pe()==mpp_root_pe())print *,'in,fv3gfs_io. time avg, time_int=',time_int
      do idx = 1,tot_diag_idx
-       if ((diag(idx)%id > 0) .or. (diag_coarse(idx)%id > 0)) then
+      requested = diag(idx)%id > 0 .or. diag_coarse(idx)%id > 0
```
Comment (oliverwm1) [inline FV3/io/FV3GFS_io.F90:3129]:
```suggestion
       requested = diag(idx)%id > 0 .or. diag_coarse(idx)%id > 0
```

**Review by oliverwm1 — APPROVED** [review]
Approving, although I am unclear on one point.

File: FV3/io/FV3GFS_io.F90 (hunk 2 above)
Comment (spencerkclark) [reply FV3/io/FV3GFS_io.F90:2997]: I set the default to `.false.` here:
https://github.com/VulcanClimateModeling/fv3gfs-fortran/blob/be96c105bddc89309d2779f807e5184a10c8b5b5/FV3/gfsphysics/GFS_layer/GFS_diagnostics.F90#L41

File: FV3/gfsphysics/GFS_layer/GFS_diagnostics.F90 (hunk 1 above)
Comment (spencerkclark) [reply FV3/gfsphysics/GFS_layer/GFS_diagnostics.F90:3271]: Thanks, good catch; this was a find and replace error

**Review by rheacangeo — APPROVED** [review]
lgtm!
